### PR TITLE
added new details column in problemstable UI

### PR DESCRIPTION
**Pull request by abhijan1102**
We need to add a new column called "Details" in the ProblemsTable with will display More... if the issue has a KB article associated. Otherwise it will remain blank. This column display will be under feature flag in instant-on portal so added a variable to toggle the column view.

### CODE REVIEW

File: src/components/ProblemsTable.tsx [hunk 1]
```diff
@@ -281,30 +286,15 @@ export const ProblemsTable = ({
...
-            const [errorId, groupCount] = row.row.original.issueid
-              ? row.row.original.issueid.split(' ', 2)
-              : [undefined, undefined];
+            const [errorId] = row.row.original.issueid ? row.row.original.issueid.split(' ', 2) : [undefined];
             return (
               <div>
                 {(row.row.subRows.length === 0 &&
                   context?.currentTable &&
                   tableStyleAccessor[context?.currentTable] === tableStyleAccessor.issueId) ||
-                !errorId ? (
```
Comment (snehaInmdr) [inline src/components/ProblemsTable.tsx:292]: We need the errorId links to show up for a percentage of users so this should not be fully removed. Its visibility should be based on the if feature flags is true or false.

Comment (abhijan1102) [reply src/components/ProblemsTable.tsx:292]: added proper toggling condition

**Review by snehaInmdr — APPROVED** [review]

File: src/components/ProblemsTable.tsx [hunk 2]
```diff
@@ -369,6 +374,41 @@ export const ProblemsTable = ({
             );
           },
         },
+        displayDetailsColumn
```
Comment (simihartstein) [inline src/components/ProblemsTable.tsx:377]: This column's visibility should be hidden using the initialState.hiddenColumns property. Here is an example: https://github.com/TanStack/table/issues/1804#issuecomment-568805891

File: src/components/ProblemsTable.tsx [hunk 3]
```diff
@@ -410,7 +450,7 @@ export const ProblemsTable = ({
           cellClassName: 'isr-problems-message',
           Cell: ({ value }: CellProps<TableRow>) => <ClampWithTooltip>{value}</ClampWithTooltip>,
         },
-      ] as Column<TableRow>[],
+      ].filter(Boolean) as Column<TableRow>[],
```
Comment (simihartstein) [inline src/components/ProblemsTable.tsx:453]: This shouldn't be necessary once you toggle visibility with the initialState.

File: src/components/ProblemsTable.tsx (hunk 2 above)
Comment (simihartstein) [reply src/components/ProblemsTable.tsx:377]: Another option someone else suggested would be to extend/augment the column type with an isVisible property, and filter it like how you are doing it below, except based on the value of the new isVisible property. https://github.com/TanStack/table/issues/1804#issuecomment-1547841355

File: src/components/ProblemsTable.tsx [hunk 4]
```diff
@@ -35,6 +35,8 @@ type Report = {
 
 type Status = 'positive' | 'warning' | 'negative' | undefined;
 
+type DisplayDetailsColumn = true | false;
```
[the remark this answers is not in the document]
Comment (abhijan1102) [reply src/components/ProblemsTable.tsx]: removed and used simple boolean

File: src/components/ProblemsTable.tsx [hunk 5]
```diff
@@ -369,6 +374,41 @@ export const ProblemsTable = ({
             );
           },
         },
+        displayDetailsColumn
+          ? {
+              id: 'details',
+              accessor: 'details',
```
[the remark this answers is not in the document]
Comment (abhijan1102) [reply src/components/ProblemsTable.tsx]: removed accessor

File: src/components/ProblemsTable.tsx [hunk 6]
```diff
@@ -369,6 +374,41 @@ export const ProblemsTable = ({
...
+              Cell: (row: CellProps<Report>) => {
+                const [errorId] = row.row.original.issueid ? row.row.original.issueid.split(' ', 2) : [undefined];
+                return (
+                  <div>
+                    {(row.row.subRows.length === 0 &&
+                      context?.currentTable &&
+                      tableStyleAccessor[context?.currentTable] === tableStyleAccessor.issueId) ||
+                    !errorId ? (
+                      ''
+                    ) : hasHelpArticle(errorId) ? (
```
[the remark this answers is not in the document]
Comment (abhijan1102) [reply src/components/ProblemsTable.tsx]: formatted

File: src/components/ProblemsTable.tsx (hunk 2 above)
Comment (abhijan1102) [reply src/components/ProblemsTable.tsx]: Used hiddenColumns

**Review by simihartstein — APPROVED** [review]
Looks good. A couple things to note:

This option will only be accessible to host applications using component composition to add the ProblemsTable. In other words, they can't use the default Report component and also set this property. This won't be a problem for InstantOn, but might require some bigger changes to work in BIC or the desktop synchronizer.

If I were to implement this, I would have used something like an array parameter to hide columns using an enum or custom type. Even if we only support hiding the details column now, this would give us the flexibility to hide more columns in the future without breaking changes. It might be worth considering in the official release.

**Comment by abhijan1102** [comment]
UI for those who do not fall under the percentage of rollout - OLD UI

UI for those who fall under the percentage of rollout - 
NEW UI

**Review by Aidas-Vaiksnoras — APPROVED** [review]

**Pull request merged**